packages/core/upload/server/register.js

```diff
@@ -60,6 +62,25 @@ const createProvider = (config) => {
     );
   }
 
+  if (providerOptions?.sizeLimit) {
+    // TODO V5: remove sizeLimit from providerOptions
+    process.emitWarning(
+      `[deprecated] In future versions, "sizeLimit" argument will be ignored from upload.config.providerOptions. Move it to upload.config`
+    );
+  }
+
+  if (!providerInstance.checkFileSize) {
```

Comment:
You should just implement that function in the baseProvider. It will be overwritten automatically by the assign below :)